store/src/java/com/zimbra/util/ExternalVolumeInfoHandler.java

```diff
@@ -24,13 +34,25 @@
  */
 public class ExternalVolumeInfoHandler {
 
+    public static final String DIR_PATH = File.separator + "opt" + File.separator + "zimbra" + File.separator + "smconfig";
+    public static final String PREV_FILE_NAME = "zsesc_prev.json";
+    public static final String PREV_FILE_PATH = DIR_PATH + File.separator + PREV_FILE_NAME;
+    public static final String CURR_FILE_NAME = "zsesc_curr.json";
+    public static final String CURR_FILE_PATH = DIR_PATH + File.separator + CURR_FILE_NAME;
+
     /**
      * LDAP Provision instance
      */
     private Provisioning provisioning;
 
     public ExternalVolumeInfoHandler(Provisioning provisioning) {
         this.provisioning = provisioning;
+        try {
+            writeServerJSONFile(provisioning.getLocalServer().getServerExternalStoreConfig(), DIR_PATH, PREV_FILE_PATH);
```

Comment:
why we are overriding files every time? consider a case this class can be used from multiple places and contracture is getting called from multiple places.

Reply:
This is done in-order to handle one corner case --> Assume our build is fresh build and its doesnt have any directory like "/opt/zimbra/smconfig/*", so whenever we are doing this operation it creates new files strictly if not existing.
This will not harm any data if the files are already present

Reply:
There are two different things.
1. preparing directories
2. saving files.
separate these things, don't add confusing names and over responsibilities to a single method.
and as discussed earlier directory path should be `/opt/zimbra/config/sm/`

Reply:
Directory path changes to /opt/zimbra/config/sm

Removed preparing directories logic from constructor this is not needed now as I removed setExternalParam function. External directories and params will get initialized only in redo operations